File: COMP_HW1/COMP_HW1_cpp/PhoneBook.cpp
#include "PhoneBookManager.h"
#include <stdlib.h>
#include <sstream>
#include <time.h>
using namespace std;


int main(int argc, char **argv){

	PhoneBookManager phonebooklist;
	int key1, key2, i; //i is just for index
	string input, name, firstname, lastname, phonenumber;

	while(true){
	  cout << "CP-2015-18634>";
	  getline(cin, input);

	  if (input.compare("")==0) {
	    cout << "Phone Book" << endl;
	    cout << "1. Add person" << endl;
	    cout << "2. Remove person" << endl;
	    cout << "3. Print phone book" << endl;
	    cout << "CP-2015-18634>";

	    cin >> key1;
	    getline(cin, input); //Consumes '\n'

	    switch(key1){
	      case 1: { //1. Add person
		    cout << "Select Type" << endl;
		    cout << "1. Person" << endl;
		    cout << "2. Work" << endl;
		    cout << "3. Family" << endl;
		    cout << "4. Friend" << endl;
		    cout << "CP-2015-18634>";
		    cin >> key2;
		    getline(cin, input); //Consumes '\n'

		    cout << "Name: ";
		    getline(cin, name);

		    //remove spaces before first name
		    i = 0;
		    while(name[i] == ' '){
		    	name.erase(i, 1);
		    }

		    firstname = name.substr(0, name.find(' '));
		    lastname = name.substr(name.find(' ')+1, name.size());

		    cout << "Phone_number: ";
		    getline(cin, phonenumber);

		    //remove all spaces in string
		    while(phonenumber.find(' ') != string::npos)
		      phonenumber.erase(phonenumber.find(' '), 1);

		    //delete '-' in phoneNumber to convert it to integer type
		    phonenumber.erase(phonenumber.find('-'), 1);
		    phonenumber.erase(phonenumber.find('-'), 1);

		    phonebooklist.addCmd(key2, firstname, lastname, atoi(phonenumber.c_str()));

	 	    break;
	      }
	      case 2: { //2. Remove person
		    int index;
		    cout << "Enter index of person: ";
		    cin >> index;
		    getline(cin, input); //Consumes '\n'

		    phonebooklist.removeCmd(index);

		    break;
	      }
	      case 3: { //3. Print phone book

		    phonebooklist.printCmd();

		    break;
	      }
	    }
	  } if (input.compare("exit")==0) {
		cout << "Bye." << endl;
		break;
	  } if(input.compare("exit") * input.compare("") != 0)
		  cout << "Press enter to go next step!" << endl;
	}

	return 0;
}

File: COMP_HW1/COMP_HW1_java/PhoneBook.java
import java.util.Scanner;

public class PhoneBook {
	public static void main(String[] args) {
		
		Scanner scan = new Scanner(System.in);
		int key1, key2, i;
		String input, name, phoneNumber;
		String[] nameSplit; //String array for first name and last name
		
		PhoneBookManager phoneBook = new PhoneBookManager();
		
		while(true){
			System.out.print("CP-2015-18634>");
			input = scan.nextLine();
			
			if(input.equals("")){
				System.out.println("Phone Book");
				System.out.println("1. Add person");
				System.out.println("2. Remove person");
				System.out.println("3. Print phone book");
				System.out.print("CP-2015-18634>");
				key1 = scan.nextInt();
				scan.nextLine(); //Consumes '\n'
				
				switch(key1){
				case 1: //1. Add person
					System.out.println("Select Type");
					System.out.println("1. Person");
					System.out.println("2. Work");
					System.out.println("3. Family");
					System.out.println("4. Friend");
					System.out.print("CP-2015-18634>");
					
					key2 = scan.nextInt();
					scan.nextLine(); //Consumes '\n'
					
					System.out.print("Name: ");
					name = scan.nextLine();
					
					//Remove blank before the first name
					for(i=0; i<name.length(); i++){
						if(name.charAt(i) != ' '){
							name = name.substring(i);
							break;
						}
					}
					//Split name to first name and last name
					nameSplit = name.split("\\s");
					
					System.out.print("Phone_number: ");
					
					//remove space or '-' in string value of phone number
					phoneNumber = scan.nextLine().replaceAll("(-|\\s+)", "");
					
					phoneBook.addCmd(key2, nameSplit[0], nameSplit[1], Integer.parseInt(phoneNumber));
					
					break;
					
				case 2: //2. Remove person
					System.out.print("Enter index of person: ");
					int index = scan.nextInt();
					scan.nextLine(); //Consumes '\n'
					
					phoneBook.removeCmd(index);
					
					break;
					
				case 3: //3. Print phone book
					phoneBook.print();
					break;
				}
			}else if(input.equals("exit")){
				System.out.println("Bye.");
				break;
			}else{
				System.out.println("Press enter to go next step!");
			}
		}
		
		scan.close();
		
	}
}
